uniflow/op/model/model_server.py

```diff
@@ -769,3 +770,223 @@ def __call__(self, data: List[str]) -> List[str]:
             inference_data.append(self.invoke_bedrock_model(prompt=d))
         data = self._postprocess(inference_data)
         return data
+
+
+class SageMakerModelServer(AbsModelServer):
+    """
+    SageMaker Model Server Class.
+
+    The AWS client authenticates by automatically loading credentials as per the methods outlined here:
+    https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html
+
+    If you wish to use a specific credential profile, please provide the profile name from your ~/.aws/credentials file.
+
+    Make sure that the credentials or roles in use have the necessary policies for SageMaker service access.
+    """
+
+    def __init__(self, model_config: Dict[str, Any]) -> None:
+        try:
+            # import in class level to avoid installing boto3
+            import boto3
+
+            super().__init__(model_config)
+            self._model_config = SageMakerModelConfig(**self._model_config)
+
+            # If user specifies profile in model config, use that profile
+            if "aws_profile" in model_config:
+                aws_profile = self._model_config.aws_profile
+                session = boto3.Session(profile_name=aws_profile)
+            # Otherwise if the user specifies credentials directly in the model config, use those credentials
+            elif (
+                self._model_config.aws_access_key_id
+                and self._model_config.aws_secret_access_key
+            ):
+                session = boto3.Session(
+                    aws_access_key_id=self._model_config.aws_access_key_id,
+                    aws_secret_access_key=self._model_config.aws_secret_access_key,
+                    aws_session_token=self._model_config.aws_session_token,
+                )
+                warnings.warn(
+                    "Using AWS credentials directly in the model config is not recommended. "
+                    "Please use a profile instead."
+                )
+            else:
+                session = boto3.Session(profile_name="default")
+                warnings.warn(
+                    "Using default profile to create the session. "
+                    "Please pass the profile name in the model config."
+                )
+
+            aws_region = (
+                self._model_config.aws_region if self._model_config.aws_region else None
+            )
```

Comment:
nit: can we refactor this part for both sagemaker and bedrock to improved re-usability.